### ISSUE TITLE
Holidays.available_regions raises NameError: uninitialized constant Holidays::REGIONS

### ISSUE THREAD

**Issue by filipegorges** [holidays/holidays#234]
Hello, 
Upon calling `Holidays.available_regions` on a console, it raises a  `NameError: uninitialized constant Holidays::REGIONS`, am I doing something wrong?

Thanks!

**Comment by ptrimble** [comment]
Hi @filipegorges! This was recently addressed in this issue: https://github.com/holidays/holidays/issues/230

Unfortunately I made a mistake with the recent v5.0.0 release and removed the ability to 'load all' automatically, not thinking that perhaps some people would be relying on it. This was my mistake, I apologize!

I have added a `load_all` method with this PR: https://github.com/holidays/holidays/pull/233. This is not released yet but will be in a new minor version this week. Calling it will load everything and give you what you want.

An issue is that I haven't considered `Holidays::REGIONS` as covered by semver simply because it is auto-generated. This might be a mistake, though. Do we think that `REGIONS` should be covered? If so then I should set it up so that, even if we do not call `load_all` that `Holidays::REGIONS` will work.

@espen or @ttwo32, thoughts?

**Comment by espen** [comment]
I'm fine with running `load_all`. But readme.md needs to be updated as it currently uses an example that will raise an error. Another solution is that either 1) `available_regions` runs `load_all` if no regions are loaded. In my case that would be expected behaviour as per earlier usage. 2) `available_regions` returns an empty array or nil do avoid the error.

**Comment by ptrimble** [comment]
@espen I definitely need to update the README, that's on me. Sorry!

To your second suggestion, I'm leaning towards having `available_regions` run `load_all` in all cases. From a user's perspective it makes sense that by running `available_regions` I am asking the gem to tell me about **all** of the regions that it knows about. Right now that is only accomplished by loading everything (unless we come up with a less memory intensive shortcut in the future).

Does this make sense? @ttwo32 Also looking for your feedback.

**Comment by espen** [comment]
I'm fine with it loading everything. Could another solution be to just return all the region symbols such as listed here: https://github.com/holidays/holidays/blob/7e061ea30e920d93d1bfb39eaf1b212973316199/lib/generated_definitions/REGIONS.rb#L3, and not actually load them ? Might be confusing though.

Anyways, I think the important thing is that it's possible to load_all and that the method doesn't fail by calling it.

**Comment by ttwo32** [comment]
@ptrimble @espen I agree with updating README , too.
I think it's better to return empty or nil or raise customize error msg(ex. 'No regions are loaded.') unless we come up with la less memory consumption.

**Comment by ptrimble** [comment]
Hey all,

I've pushed up a PR that I think addresses this. I went with @espen's original idea, which is to require the 'REGIONS' file only. This way we report to the user what we have available and, when they actually call it, we load it.

I'm not 100% sure I want this behavior going forward but the main point that I wanted to nail down was that we accurately reported the region symbols we supported by calling `Holidays.available_regions`. This seems to be the fasted way to do it. We don't have to actually call `load_all` like I said originally. We can just return the array and then when a region is provided we load it, but only then.

Let me know what you think! If I don't hear back I'll merge this and get out a new version this weekend.
